controllers/tinkerbellmachine_controller.go

```diff
@@ -54,30 +56,79 @@ type TinkerbellMachineReconciler struct {
 // +kubebuilder:rbac:groups=bmc.tinkerbell.org,resources=jobs,verbs=get;list;watch;create
 
 // Reconcile ensures that all Tinkerbell machines are aligned with a given spec.
-func (tmr *TinkerbellMachineReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	bmrc, err := tmr.newReconcileContext(ctx, req.NamespacedName)
```

Comment:
The implementation of the `newReconcileContext()` has been moved to the `Reconcile()` call (lines 64-91).